### ISSUE TITLE
react-router cli help menu is no longer accessible after upgrading to @react-router/dev@7.1.0

### ISSUE THREAD

**Issue by IsaacCloos** [remix-run/react-router#12627]
### I'm using React Router as a...

framework

### Reproduction

upgrade to `npm i -D @react-router/dev`

try

`npx react-router -?`
`npx react-router -h`
`npx react-router --help`
`npx react-router help` // this command says 'React Router Vite plugin not found in Vite config' despite my vite config including the plugin? Little extra discovery, not really the main point of this issue though.
`npx react-router dev -h`

all throw an error

### System Info

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (12) x64 13th Gen Intel(R) Core(TM) i7-1355U
    Memory: 6.62 GB / 15.58 GB
  Binaries:
    Node: 22.12.0 - C:\Program Files\nodejs\node.EXE
    npm: 11.0.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (117.0.2045.31)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @react-router/dev: ^7.1.0 => 7.1.0
    @react-router/fs-routes: ^7.1.0 => 7.1.0
    @react-router/node: ^7.1.0 => 7.1.0
    @react-router/serve: ^7.1.0 => 7.1.0
    react-router: ^7.1.0 => 7.1.0
    vite: ^5.4.11 => 5.4.11
```

### Used Package Manager

npm

### Expected Behavior

-?
-h
--help
help

all should work for general or command specific information

### Actual Behavior

node_modules\arg\index.js:132
                                                throw new ArgError(
                                                ^

ArgError: unknown or unexpected option: -?
    at arg (C:\...\node_modules\arg\index.js:132:13)
    at Object.<anonymous> (C:\...\node_modules\@react-router\dev\bin.js:8:12)
    at Module._compile (node:internal/modules/cjs/loader:1565:14)
    at Object..js (node:internal/modules/cjs/loader:1708:10)
    at Module.load (node:internal/modules/cjs/loader:1318:32)
    at Function._load (node:internal/modules/cjs/loader:1128:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:219:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'ARG_UNKNOWN_OPTION'
}

**Comment by timdorr** [comment]
Duplicate of #12606

**Comment by PhilDL** [comment]
Ah yes, my bad, I totally forgot you could use `react-router` without any options in the fix I proposed in #12609 . It was not covering that case and I wrongly assumed user would always do `react-router build | dev | reveal....`.

The fix in that case would be to set the `permissive` flag in the `arg()` to allow any kind of unknown options for that specific file (since it's only used to set an ENV variable it should be safe). 

```diff
diff --git a/bin.js b/bin.js
index 0373bf96f7c42f5460dc2e9fea394e9a7e947e06..023600393b2b3f5515510016cc839134be210bfd 100755
--- a/bin.js
+++ b/bin.js
@@ -5,7 +5,7 @@ let arg = require("arg");
 // default `NODE_ENV` so React loads the proper version in it's CJS entry script.
 // We have to do this before importing `run.ts` since that is what imports
 // `react` (indirectly via `react-router`)
-let args = arg({}, { argv: process.argv.slice(2), stopAtPositional: true });
+let args = arg({}, { argv: process.argv.slice(2), permissive: true });
 if (args._[0] === "dev") {
   process.env.NODE_ENV = process.env.NODE_ENV ?? "development";
 } else {
```

@timdorr sorry for the inconvenience, this issue is different than the other one. I don't have time on this christmas eve to do a PR with these changes, but I believe this is non urgent, as the important commands can still be used.
